modules/c1xBidAdapter.js

```diff
@@ -0,0 +1,142 @@
+var CONSTANTS = require('src/constants.json');
+var utils = require('src/utils.js');
+var bidfactory = require('src/bidfactory.js');
+var bidmanager = require('src/bidmanager.js');
+var adloader = require('src/adloader');
+var adaptermanager = require('src/adaptermanager.js');
+
+/**
+ * Adapter for requesting bids from C1X header tag server.
+ * v2.0 (c) C1X Inc., 2017
+ *
+ * @param {Object} options - Configuration options for C1X
+ *
+ * @returns {{callBids: _callBids}}
+ * @constructor
+ */
+function C1XAdapter() {
+  // default endpoint. Can be overridden by adding an "endpoint" property to the first param in bidder config.
+  var ENDPOINT = 'http://ht-integration.c1exchange.com:9000/ht',
+    PIXEL_ENDPOINT = '//px.c1exchange.com/pubpixel/',
+    PIXEL_FIRE_DELAY = 3000,
+    LOG_MSG = {
+      invalidBid: 'C1X: ERROR bidder returns an invalid bid',
+      noSite: 'C1X: ERROR no site id supplied',
+      noBid: 'C1X: INFO creating a NO bid for Adunit: ',
+      bidWin: 'C1X: INFO creating a bid for Adunit: '
+    },
+    BIDDER_CODE = 'c1x';
+
+  var pbjs = window.pbjs;
+
+  pbjs._c1xResponse = function(c1xResponse) {
+    if (c1xResponse) {
+      var response = c1xResponse;
+
+      if (typeof c1xResponse === CONSTANTS.objectType_string) {
+        response = JSON.parse(c1xResponse);
```

Comment:
Please wrap `JSON.parse` in a `try/catch` block to catch any parsing exceptions. See `appnexusAstBidAdapter` or other adapters for examples of this if needed